### PR TITLE
Update docs about complex character support

### PR DESCRIPTION
**Pull request by justinmc**
## Description

Support for grapheme clusters and surrogate pairs was added to Flutter in https://github.com/flutter/flutter/pull/59267.  However, some documentation was found that was out of date with that change, so this PR updates it.

## Related Issues

Discovered by @LongCatIsLooong in https://github.com/flutter/flutter/pull/66567#discussion_r494510353.

## Tests

None, docs change.

### CODE REVIEW

File: packages/flutter/lib/src/services/text_formatter.dart [hunk 1]
```diff
@@ -316,31 +316,29 @@ class LengthLimitingTextInputFormatter extends TextInputFormatter {
...
-  /// characters. If you expect to encounter these kinds of characters, be
-  /// generous in the maxLength used.
+  /// For a specific definition of what is considered a character, see the
+  /// [characters](https://pub.dev/packages/characters) package on Pub, which is
+  /// what Flutter uses to delineate characters. In general, even complex
+  /// characters like surrogate pairs and extended grapheme clusters are
+  /// correctly interpreted by Flutter as each being a single user-perceived
+  /// character.
   ///
   /// For instance, the character "ö" can be represented as '\u{006F}\u{0308}',
```
Comment (justinmc) [inline packages/flutter/lib/src/services/text_formatter.dart:332]: I debated whether to drop these two paragraphs altogether, thinking that they might be more confusing than just removing them, but I decided to keep them.  Let me know if you think they should go.

**Review by LongCatIsLooong — APPROVED** [review]
Thank you! The material textfield and the cupertino text field have similar docs that need updating too I think.

File: packages/flutter/lib/src/services/text_formatter.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/services/text_formatter.dart:332]: Personally I think it helps clarify what the problem is without involving the terrifying terms.

**Comment by justinmc** [comment]
Ah good catch.  I've updated those two using my first change as a template.

**Pull request merged**